mne/viz/topo.py

```diff
@@ -247,39 +247,43 @@ def _check_vlim(vlim):
 
 
 def _imshow_tfr(ax, ch_idx, tmin, tmax, vmin, vmax, onselect, ylim=None,
-                tfr=None, freq=None, vline=None, x_label=None, y_label=None,
-                colorbar=False, picker=True, cmap=('RdBu_r', True), title=None,
-                hline=None):
-    """Show time-freq map on topo."""
+                tfr=None, freq=None, x_label=None, y_label=None,
+                colorbar=False, cmap=('RdBu_r', True)):
+    """Show time-freq map as 2d image."""
     import matplotlib.pyplot as plt
     from matplotlib.widgets import RectangleSelector
 
-    extent = (tmin, tmax, freq[0], freq[-1])
     cmap, interactive_cmap = cmap
+    extent = (tmin, tmax, freq[0], freq[-1])
+    times = np.linspace(tmin, tmax, num=tfr[ch_idx].shape[1])
+
+    # construct a grid of borders between time/frequency bins
+    time_diff = np.diff(times) / 2. if len(times) > 1 else [0.0005]
+    freq_diff = np.diff(freq) / 2. if len(freq) > 1 else [0.5]
+    time_lims = np.concatenate([[times[0] - time_diff[0]], times[:-1] +
+                               time_diff, [times[-1] + time_diff[-1]]])
+    freq_lims = np.concatenate([[freq[0] - freq_diff[0]], freq[:-1] +
+                               freq_diff, [freq[-1] + freq_diff[-1]]])
+    time_mesh, freq_mesh = np.meshgrid(time_lims, freq_lims)
+
+    img = ax.pcolormesh(time_mesh, freq_mesh, tfr[ch_idx], cmap=cmap,
+                        vmin=vmin, vmax=vmax)
+    ax.set_xlim(extent[0], extent[1])
+    ax.set_ylim(extent[2], extent[3])
 
-    img = ax.imshow(tfr[ch_idx], extent=extent, aspect="auto", origin="lower",
-                    vmin=vmin, vmax=vmax, picker=picker, cmap=cmap)
-    if isinstance(ax, plt.Axes):
-        if x_label is not None:
-            ax.set_xlabel(x_label)
-        if y_label is not None:
-            ax.set_ylabel(y_label)
-    else:
-        if x_label is not None:
-            plt.xlabel(x_label)
-        if y_label is not None:
-            plt.ylabel(y_label)
+    if not isinstance(ax, plt.Axes):
+        ax = plt.gca()
+    if x_label is not None:
+        ax.set_xlabel(x_label)
+    if y_label is not None:
+        ax.set_ylabel(y_label)
     if colorbar:
         if isinstance(colorbar, DraggableColorbar):
             cbar = colorbar.cbar  # this happens with multiaxes case
         else:
             cbar = plt.colorbar(mappable=img)
         if interactive_cmap:
             ax.CB = DraggableColorbar(cbar, img)
-    if title:
-        plt.title(title)
```

Comment:
As I mentioned before - I've cleaned up _imshow_tfr a bit.
`title` was not used - `tfr.plot()` sets the title itself after using _imshow_tfr (more precisely - suptitle).